### PR TITLE
Inference PP changes for neox

### PR DESCRIPTION
**Pull request by jeffra**

### CODE REVIEW

File: deepspeed/module_inject/replace_module.py [hunk 1]
```diff
@@ -263,14 +264,16 @@ def replace_with_policy(child,
...
+                                            if hasattr(child.attention,'rotary_ndims') else -1
                 transformer_config = transformer_inference.DeepSpeedInferenceConfig(
                     hidden_size=hidden_size,
                     heads=num_attention_heads,
                     layer_norm_eps=config.layer_norm_eps if hasattr(
                         config,
                         'layer_norm_eps') else (config.layer_norm_epsilon if hasattr(
                             config,
-                            'layer_norm_epsilon') else 1e-12),
+                            'layer_norm_epsilon') else 1.0e-5),
```
Comment (jeffra) [inline deepspeed/module_inject/replace_module.py:276]: Let's parameterize this during injection

File: deepspeed/module_inject/replace_policy.py [hunk 2]
```diff
@@ -204,8 +212,8 @@ def __init__(self, client_module, inference=True):
         if MegatronLayerPolicy._orig_layer_class is None:
             try:
                 import megatron
-                from megatron.model.transformer import ParallelTransformerLayer
-                MegatronLayerPolicy._orig_layer_class = ParallelTransformerLayer
+                from megatron.model.transformer import ParallelTransformerLayer1
+                MegatronLayerPolicy._orig_layer_class = ParallelTransformerLayer1
```
Comment (jeffra) [inline deepspeed/module_inject/replace_policy.py:216]: bug

File: deepspeed/module_inject/replace_policy.py [hunk 3]
```diff
@@ -214,6 +222,8 @@ def get_hidden_heads(self):
                 self.client_module.attention.num_attention_heads
 
     def attention(self):
+        print("coming to the wrong policy")
+        exit()
```
Comment (jeffra) [inline deepspeed/module_inject/replace_policy.py:226]: remove

File: deepspeed/runtime/config.py [hunk 4]
```diff
@@ -873,7 +873,7 @@ def __init__(self, config: Union[str, dict], mpu=None):
 
         self._initialize_params(self._param_dict)
         self._configure_train_batch_size()
-        self._do_sanity_check()
+#        self._do_sanity_check()
```
Comment (jeffra) [inline deepspeed/runtime/config.py:876]: we need to bring this back

File: deepspeed/runtime/config.py [hunk 5]
```diff
@@ -991,15 +991,17 @@ def _batch_assertion(self):
...
-            "to micro_batch_per_gpu * gradient_acc_step * world_size "
-            f"{train_batch} != {micro_batch} * {grad_acc} * {self.world_size}"
-        )
+
+#
+#assert train_batch == micro_batch * grad_acc * self.world_size, (
+#    f"Check batch related parameters. train_batch_size is not equal "
+#    "to micro_batch_per_gpu * gradient_acc_step * world_size "
+#    f"{train_batch} != {micro_batch} * {grad_acc} * {self.world_size}"
+#)
```
Comment (jeffra) [inline deepspeed/runtime/config.py:1004]: bring these back

File: deepspeed/runtime/engine.py [hunk 6]
```diff
@@ -706,7 +706,7 @@ def loss_scale(self):
         return self._config.loss_scale
 
     def gradient_accumulation_steps(self):
-        return self._config.gradient_accumulation_steps
+        return 1  #self._config.gradient_accumulation_steps
```
Comment (jeffra) [inline deepspeed/runtime/engine.py:709]: bring back

File: deepspeed/runtime/pipe/engine.py [hunk 7]
```diff
@@ -1323,6 +1339,10 @@ def load_module_state_dict(self, state_dict, strict=True):
             state_dict (str, None): unused
             strict (bool, optional): Strict state loading. Defaults to True.
         """
+        print(
+            f"**************** calling load form PIPE Engine {state_dict} **********************"
+        )
+
```
Comment (jeffra) [inline deepspeed/runtime/pipe/engine.py:1345]: remove

File: deepspeed/utils/nvtx.py [hunk 8]
```diff
@@ -7,6 +7,8 @@ def instrument_w_nvtx(func):
     if hasattr(torch.cuda.nvtx, "range"):
 
         def wrapped_fn(*args, **kwargs):
+            #import pdb;pdb.set_trace()
+            print(args)
```
Comment (jeffra) [inline deepspeed/utils/nvtx.py:11]: remove

**Comment by jeffra** [comment]
need to fix `-lcurand` issue with amd build before merge

**Review by RezaYazdaniAminabadi — APPROVED** [review]

**Pull request merged**